### DISCUSSION TITLE
Running brew livecheck on Linux reports a different version than on macOS

### DISCUSSION THREAD

**Discussion by cesarcoatl** [Homebrew/discussions#4099] [Linux]
### Additional Info

- [X] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.18-24-gadfd868
ORIGIN: https://github.com/Homebrew/brew
HEAD: adfd868831f4bce6d26f3962c63e4475f3781a82
Last commit: 6 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3ec56639519622adfa6a3de83ba0dd64753cb99f
Core tap last commit: 12 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CLEANUP_MAX_AGE_DAYS: 3
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_GIT_EMAIL: thecesrom@gmail.com
HOMEBREW_GIT_NAME: César Román
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: 16-core 64-bit unknown_0x6_0x8d
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.79.1-microsoft-standard-WSL2 x86_64 GNU/Linux
OS: Ubuntu 22.04.1 LTS (jammy)
WSL: 2 (Microsoft Store)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I maintain a Tap ([coatl-dev](https://github.com/coatl-dev/homebrew-coatl-dev)), and when I run `livecheck` on [`ignition@7.9`](https://github.com/coatl-dev/homebrew-coatl-dev/blob/coatl/Formula/ignition@7.9.rb) I get a different result in Linux and macOS.

Linux

```sh
$ brew livecheck ignition@7.9
ignition@7.9: 64-7 ==> 7.9.21
```

macOS

```sh
$ brew livecheck ignition@7.9
ignition@7.9: 7.9.21 ==> 7.9.21
```

So, why does it differ?

Thanks

**Comment by cho-m** [comment]
Homebrew's version detection is picking up the `64-7` from `linux-64-7.9.21` in URL. Currently, you will need to add a `version` stanza to help with the Linux URL unless the detection is changed in `brew`.

**Reply by cesarcoatl** [reply]
Thanks for replying, @cho-m.

I tried that on https://github.com/coatl-dev/homebrew-coatl-dev/pull/67, and it failed the `audit` with the following:

```text
brew audit --tap=coatl-dev/coatl-dev
==> FAILED
Full audit --tap=coatl-dev/coatl-dev output
  coatl-dev/coatl-dev/ignition@7.9:
    * Stable: version 7.9.21 is redundant with version scanned from URL
```

For CI I am using both `macos-latest` and `ubuntu-latest`. Maybe it will pass on Ubuntu, but it will still fail on macOS.

**Reply by cesarcoatl** [reply]
> ...is picking up the `64-7` from `linux-64-7.9.21` in URL.

I did end up adding a `version` stanza to the non-versioned formula `ignition` when they added `x86-64-8.1.x.zip` to the `8.1` version, as `livecheck` was returning `64-8`.

I guess my options are:

1. Let `livecheck` report the version assumed from the `url` on Linux
2. Accept PRs not passing the tests

I'd rather go with option 1.
